crates/driver/src/domain/competition/solution/trade.rs

```diff
@@ -205,6 +234,27 @@ impl Fulfillment {
         Ok(surplus)
     }
 
+    /// Returns the surplus denominated in the surplus token.
+    ///
+    /// The surplus token is a buy token for a sell order and a sell token for a
+    /// buy order.
+    ///
+    /// The surplus is defined as the difference between the executed price and
+    /// the order limit price.
+    pub fn surplus(&self, prices: ClearingPrices) -> Result<eth::Asset, Error> {
```

Comment:
```suggestion
    pub fn surplus(&self, prices: &ClearingPrices) -> Result<eth::Asset, Error> {
```
Seems like we could only take a reference everywhere where we currently require ownership.

Also unrelated to this PR it's a bit awkward that the `ClearingPrices` are strictly related to a `Trade` but this is not enforce on a type level. Also the struct is `pub` in the crate so I think it's probably possible to skrew up and use `ClearingPrices` for a trade it's not related to.
Just wanted to bring it up in case we want to refactor the `ClearingPrices` at some point.